### PR TITLE
Fix/do not use computed

### PR DESCRIPTION
**Pull request by matejrisek**
## Description

Addresses issue [described in this GitHub issue](https://github.com/hashicorp/terraform-provider-tfe/issues/552).
It should prevent sending trigger-patterns and trigger-prefixes in the same request, which is currently causing failures for the Terraform Enterprise with version `v202205-1`. The issues is occurring because this version of TFE checks for an invariant that `trigger-patterns` and `trigger-prefixes` should NOT be present at the same time. The latest Atlas codebase (and newer versions of TFE) is checking that these values (arrays) are not populated rather for the presence, meaning the issues is not present in newer versions. 

The above fix is using raw configuration reading, and for that we had to bump up the SDK version to `v2.8.0`

This PR also addresses some other quirks we were experiencing because the fields `trigger-patterns` and `trigger-prefixes` were marked as `computed`. The `computed` trait has been removed for these fields and we introduced a more comprehensive test suite.

_Remember to:_

- _Update the [Change Log](https://github.com/hashicorp/terraform-provider-tfe/blob/main/README.md#updating-the-changelog)_

- _Update the [Documentation](https://github.com/hashicorp/terraform-provider-tfe/blob/main/README.md#updating-the-documentation)_

## Testing plan

1. Either test against the `TFE` version `v202205-1` or checkout the Atlas codebase with the same git tag
2. Create a workspace that has `trigger-prefixes` set
3. The creation should NOT fail because of the presence of both `trigger-prefixes` and `trigger-patterns`

## External links

_Include any links here that might be helpful for people reviewing your PR. If there are none, feel free to delete this section._

- [GitHub Issue](https://github.com/hashicorp/terraform-provider-tfe/issues/552)

## Output from acceptance tests

_Please run applicable acceptance tests locally and include the output here. See TESTS.md to learn how to run acceptance tests._

_If you are an external contributor, your contribution(s) will first be reviewed before running them against the project's CI pipeline._

```
$ TESTARGS="-run TestAccTFEWorkspace" make testacc

...
```

### CODE REVIEW

**Review by omarismail — APPROVED** [review]
amazing set of tests! LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ENHANCEMENTS:
 * d/agent_pool: Improve efficiency of reading agent pool data when the target organization has more than 20 agent pools ([#508](https://github.com/hashicorp/terraform-provider-tfe/pull/508))
 * Added warning logs for 404 error responses ([#538](https://github.com/hashicorp/terraform-provider-tfe/pull/538))
+* r/tfe_workspace: Fix `trigger-prefixes` could not be updated because of the conflict with `trigger-patterns` in some cases - as described in this [GitHub Issue](https://github.com/hashicorp/terraform-provider-tfe/issues/552) ([#564](https://github.com/hashicorp/terraform-provider-tfe/pull/564/))
```
Comment (brandonc) [inline CHANGELOG.md:6]: Can you add this to a new Bug Fixes heading?

**Review by brandonc — APPROVED** [review]
I smoke tested this on the latest TFC and it behaved nicely, with and without the beta feature enabled.

File: CHANGELOG.md (hunk 1 above)
Comment (matejrisek) [reply CHANGELOG.md]: 👍🏻  1a89a9bd46df04ff5eb97226488f1610445b8195

**Pull request merged**